algorithms/linfa-preprocessing/src/error.rs

```diff
@@ -0,0 +1,37 @@
+//! Error definitions for preprocessing
+use thiserror::Error;
+pub type Result<T> = std::result::Result<T, Error>;
+
+#[derive(Error, Debug)]
+pub enum Error {
+    #[error("wrong measure ({0}) for scaler: {1}")]
+    WrongMeasureForScaler(String, String),
+    #[error("subsamples greater than total samples: {0} > {1}")]
+    TooManySubsamples(usize, usize),
+    #[error("not enough samples")]
+    NotEnoughSamples,
+    #[error("not a valid float")]
+    InvalidFloat,
+    #[error("minimum value for MinMax scaler cannot be greater than the maximum")]
+    FlippedMinMaxRange,
+    #[error("n_gram boundaries cannot be zero (min = {0}, max = {1})")]
+    InvalidNGramBoundaries(usize, usize),
+    #[error("n_gram min boundary cannot be greater than max boundary (min = {0}, max = {1})")]
+    FlippedNGramBoundaries(usize, usize),
+    #[error("document frequencies have to be between 0 and 1 (min = {0}, max = {1})")]
+    InvalidDocumentFrequencies(f32, f32),
+    #[error("min document frequency cannot be greater than max document frequency (min = {0}, max = {1})")]
+    FlippedDocumentFrequencies(f32, f32),
+    #[error(transparent)]
+    RegexError(#[from] regex::Error),
+    #[error(transparent)]
+    IoError(#[from] std::io::Error),
+    #[error("Encoding error {0}")]
+    EncodingError(std::borrow::Cow<'static, str>),
+    #[error(transparent)]
+    LinalgError(#[from] ndarray_linalg::error::LinalgError),
+    #[error(transparent)]
+    NdarrayLinalgEmptyError(#[from] ndarray_stats::errors::EmptyInput),
+    #[error("Eigenvectors were not computed")]
+    NoneEignenvectors,
```

Comment:
typo `NoneEigenvector`. Does this error occurs when the decomposition fails?

Reply:
In theory it should never happen. The svd function takes two booleans to know whether to build `U` and/or `V_T` in output, so these two matrices are given as `Option`. That error would only appear if someone set the needed flag to false, getting `None` for the matrix needed. So this kind of error can only appear to a developer modifying the whitening procedure. 

When decomposition fails I just propagate the `ndarray-linalg` error

Reply:
I would remove the error type and just unwrap the result with a comment explaining why this is safe